source/Ship.cpp

```diff
@@ -1104,21 +1105,27 @@ vector<string> Ship::FlightCheck() const
 	// Report the first error condition that will prevent takeoff:
 	if(IdleHeat() >= MaximumHeat())
 		checks.emplace_back("overheating!");
-	else if(energy <= 0.)
+	else if(energy <= 0. && !docked)
 		checks.emplace_back("no energy!");
-	else if((energy - consuming <= 0.) && (fuel <= 0.))
+	else if((energy - consuming <= 0.) && (fuel <= 0.) && !docked)
 		checks.emplace_back("no fuel!");
-	else if(!thrust && !reverseThrust && !afterburner)
+	else if(!thrust && !reverseThrust && !afterburner && !docked)
 		checks.emplace_back("no thruster!");
-	else if(!turn)
+	else if(!turn && !docked)
 		checks.emplace_back("no steering!");
 
 	// If no errors were found, check all warning conditions:
 	if(checks.empty())
 	{
-		if(!thrust && !reverseThrust)
+		if(energy <= 0.)
+			checks.emplace_back("no energy?");
+		if(!thrust && !afterburner && !reverseThrust)
+			checks.emplace_back("no thruster?");
+		if(!turn)
+			checks.emplace_back("no steering?");
```

Comment:
We could use a warning for energy - consuming <= 0. I think. (the no fuel! from above had that check)
Or is it not included for a specific reason?

Also we can optimize this a bit by only checking for these added conditions if the ship is a docked ship.

Reply:
I've added a new check for `(energy - consuming <= 0.) && (fuel <= 0.)` which matches the previous version. Not sure if you wanted this or just a general warning about ships using fuel generators only; that could be added after the `fuel?` check.

Reply:
I would have been fine with either, you can mark this as resolved